core/src/main/java/org/elasticsearch/discovery/zen/UnicastZenPing.java

```diff
@@ -292,6 +292,7 @@ public void clearTemporalResponses() {
      */
     @Override
     public void ping(final PingListener listener, final TimeValue duration) {
+        final List<DiscoveryNode> sortedNodesToPing = buildNodesToPing();
```

Comment:
Hi @bleskes, I have updated this base on your suggestion, Could you review again? Thanks